lib/seattleflu/id3c/cli/command/etl/fhir_customizations.py

```diff
@@ -0,0 +1,148 @@
+
+def update_details_nwgc_id(sample: Any, nwgc_id: str, additional_details: dict) -> None:
+    """
+    Given a *sample* fetched from `warehouse.sample`,
+    extend `sample.details.nwgc_id` to an array if needed.
+
+    Add provided *nwgc_id* to the existing array if it doesn't already exist
+    and add to the *additional_details* dict.
+    """
+    if not sample.details:
+        return
+
+    existing_nwgc_ids = sample.details.get("nwgc_id", [])
+
+    # Extend details.nwgc_id to an array
+    if not isinstance(existing_nwgc_ids, list):
+        existing_nwgc_ids = [existing_nwgc_ids]
+
+    existing_nwgc_ids.append(int(nwgc_id))
+
+    # Concatenate exisiting and new nwgc_ids and deduplicate
+    additional_details["nwgc_id"] = list(set(existing_nwgc_ids))
+
+
+def process_result(db: DatabaseSession,
+                   sample_id: int,
+                   observation: Observation,
+                   report_effective_datetime: str = None) -> None:
+    """
+    Given an  *observation* containing a presence-absence test result,
+    upserts them to ID3C, attaching a sample and target ID.
+
+    Includes handling of custom extensions that are expected to be in the
+    *observation* resource.
+    """
+    target_identifier = matching_system_code(observation.code, f"{INTERNAL_SYSTEM}/target/identifier")
+    target_control_status = matching_system_code(observation.code, f"{INTERNAL_SYSTEM}/target/control")
+
+    assert target_identifier, \
+        f"No {INTERNAL_SYSTEM}/target/identifier code found for observation"
+
+    assert target_control_status, \
+        f"No {INTERNAL_SYSTEM}/target/control code found for observation"
+
+    target = find_or_create_target(db,
+        identifier = target_identifier,
+        control = target_control_status
+    )
+
+    result_identifier = identifier(observation, f"{INTERNAL_SYSTEM}/presence-absence")
+
+    assert result_identifier, \
+        f"No {INTERNAL_SYSTEM}/presence-absence identifier found for observation"
+
+    result_value = observation_value(observation)
+
+    details = { "device": observation.device.identifier.value }
+
+    if report_effective_datetime:
+        details["assay_date"] = report_effective_datetime.as_json()
+
+    expected_extensions = [
+        ("assay_type", "valueString"),
+        ("replicates", "valueString"),
+        ("extraction_date", "valueDate")
+    ]
+
+    for extension, value_type in expected_extensions:
+        extension_value = matching_extension_value(
+            extensions = observation.extension,
+            url = f"{INTERNAL_SYSTEM}/{extension}",
+            value_type = value_type
+        )
+
+        if extension_value:
+            try:
+                details[extension] = extension_value.as_json()
+            except AttributeError:
+                try:
+                    details[extension] = json.loads(extension_value)
+                except json.decoder.JSONDecodeError:
+                    details[extension] = extension_value
```

Comment:
Could you comment on why we try these three approaches in this order to parse an `extension_value`?